ASFAchievementManager/AchievementHandler.cs

```diff
@@ -254,9 +234,9 @@ public sealed class AchievementHandler : ClientMsgHandler {
 			List<CMsgClientStoreUserStats2.Stats> statsToSet = new List<CMsgClientStoreUserStats2.Stats>();
 
 			if (achievements.Count == 0) { //if no parameters provided - set/reset all. Don't kill me Archi.
-				for (int counter = 0; counter < Stats.Count; counter++) {
-					if (!Stats[counter].Restricted) {
-						SetStat(statsToSet, Stats, response, counter, set);
+				foreach (StatData stat in Stats) {
```

Comment:
Use `foreach (StatData stat in Stats.Where(s => !s.Restricted))` and remove the `if`-statement inside the loop